.github/workflows/on-pull.yml

```diff
@@ -26,29 +24,20 @@ jobs:
           fetch-depth: 0
       - uses: wagoid/commitlint-github-action@v3
 
-  install:
+  install-lint-build:
     runs-on: ubuntu-latest
-    name: Dependency installation
-    needs: one_run
+    name: Install, Lint, Build
+    needs: cancel-previous-runs
     steps:
       - uses: actions/checkout@v4.1.1
+      - uses: actions/cache@v3
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
       - uses: ./.github/actions/nodejs
         name: Install dependencies
-
-  lint:
-    runs-on: ubuntu-latest
-    name: Lint code
-    needs: install
-    steps:
-      - uses: actions/checkout@v4.1.1
-      - uses: ./.github/actions/nodejs
       - run: yarn run lint
-
-  build:
-    runs-on: ubuntu-latest
-    name: Build and bundle size check
-    needs: install
-    steps:
-      - uses: actions/checkout@v4.1.1
-      - uses: ./.github/actions/nodejs
       - uses: ./.github/actions/build
```

Comment:
I think we could even replace this with nx commands directly and remove actions/build